stack-graphs/src/stitching.rs

```diff
@@ -719,9 +721,13 @@ impl PathStitcher {
 /// [`find_all_complete_partial_paths`]: #method.find_all_complete_partial_paths
 pub struct ForwardPartialPathStitcher {
     candidate_partial_paths: Vec<Handle<PartialPath>>,
-    queue: VecDeque<PartialPath>,
-    next_iteration: VecDeque<PartialPath>,
-    cycle_detector: CycleDetector<PartialPath>,
+    queue: VecDeque<(PartialPath, AppendingCycleDetector<OwnedOrDatabasePath>)>,
+    next_iteration: (
+        VecDeque<PartialPath>,
+        VecDeque<AppendingCycleDetector<OwnedOrDatabasePath>>,
+    ),
```

Comment:
This is a particularly ugly bit. The main reason for this is that the C API directly exposes `next_iteration`. Without that requirement, I would have a `VecDeque<(_, _)>` here. One way to get this is by introducing an intermediate type in the C API for the elements of `next_iteration`, of which the first field would be the path, and the rest some abstract internal state (enough to make the memory layout work).

Reply:
I think that is worth considering, but would be best handled in a follow-on PR if you do tackle it.  Maybe add a comment clarifying why it's a tuple of vecdeques instead of a vecdeque of tuples.

I also think it's okay to keep it like this — we're already doing a lot of "struct of arrays" instead of "arrays of structs" in how we do arena stuff, so it's at least conceptually aligned with the rest of the crate.

Reply:
Let's keep it as-is for now, then. It is purely internal and not visible in the API AFAIK.